### ISSUE TITLE
Question: Recommended Way to Listen to UnreadCount Updates?

### ISSUE THREAD

**Issue by emgbtg** [GetStream/stream-chat-swift#1376]
> ## What did you do?
> Trying to implement unread subscription per: https://getstream.io/chat/docs/ios-swift/unread/?language=swift
> 
> ## What did you expect to happen?
> Custom CurrentUserDelegate with extra data would be called on unread updates 
> 
> ## What happened instead?
> My custom delegate was not called
> 
> ## GetStream Environment
> **GetStream Chat version:**
> 4.0.0 beta 6 
> ## Additional context

Im using a custom ChatChannelListVC with extra data types. I would like to know when the channel list has updated its unread count in order to show/hide a Tab Bar badge indicating that you have a new unread message. After reading the docs I saw I needed to set  a custom CurrentUserDelegate since I am using extra data:

`          

           let channelDelegate = MyChatChannelControllerDelegate()
           
           chatClient.currentUserController().setDelegate(channelDelegate)
           
            chatClient.currentUserController().synchronize { error in
                if let error = error {
                    /// 4: Handle possible errors
                    print(error)
                }
            }

// Delegate implementation

`   

    public class MyChatChannelControllerDelegate: _CurrentChatUserControllerDelegate {
    
    public typealias ExtraData = MyStreamExtraDataType

    public func currentUserController(_ controller: _CurrentChatUserController<MyStreamExtraDataType>, didChangeCurrentUserUnreadCount: UnreadCount) {
        print(didChangeCurrentUserUnreadCount.messages)
    }

    public func currentUserController(_ controller: _CurrentChatUserController<MyStreamExtraDataType>, didChangeCurrentUser: EntityChange<_CurrentChatUser<MyStreamExtraDataType>>) {

    }
}`

After stepping through some breakpoint I found that _CurrentChatUserAvatarView is setting itself as the controllers delegate ( CurrentChatUserAvatarView.swift line 24). Could this be overriding my call to setDelegate?

**Comment by dmigach** [comment]
Hello @emgbtg ! In the 4.0-beta 9 there was a major rework of everything related to extra data.
Would it be possible for you to adopt the freshest beta-11 and check if it helps with your issue?
If you need help with migrating to the new extra data API, release notes for beta-9 have some explanations 
https://github.com/GetStream/stream-chat-swift/releases/tag/4.0.0-beta.9

**Comment by emgbtg** [comment]
Thank you @tbarbugli for the quick response. I switched to Beta.11 branch and still seeing this issue. I did find a somewhat hacky workaround that works in both beta.6 and beta.11:

`
    override public func setUp() {
         super.setUp()
         userAvatarView.controller?.setDelegate(channelDelegate)

         chatClient.currentUserController().setDelegate(channelDelegate)
         chatClient.currentUserController().synchronize()
     }
`

In my ChatChannelListVC if I override setUp() and set the userAvatar's controller delegate to my custom delegate it works. Again, I think this may be a bug where userAvatarView is setting the delegate when it sets its controller (CurrentChatUserAvatarView ln 17 in beta.11)
